### PR TITLE
Fix: Turn Video into immutable object

### PR DESCRIPTION
**Pull request by localheinz**
This PR

* [x] turns `Video` into an immutable object 

Follows #56.

### CODE REVIEW

File: src/Component/Video/Video.php [hunk 1]
```diff
@@ -240,152 +174,257 @@ public function galleryLocation()
...
-    public function addTag(TagInterface $tag)
+    /**
+     * @param float $rating
+     *
+     * @return static
+     */
+    public function withRating($rating = null)
     {
-        Assertion::lessThan(count($this->tags), VideoInterface::TAG_MAX_COUNT);
+        Assertion::numeric($rating);
```
Comment (alucic) [inline src/Component/Video/Video.php]: `$rating` should probably be required param

**Comment by localheinz** [comment]
@alucic 

Amended!

File: src/Component/Video/Video.php [hunk 2]
```diff
@@ -394,41 +433,59 @@ private function setRequiresSubscription($requiresSubscription = null)
 
         Assertion::nullOrChoice($requiresSubscription, $choices);
```
Comment (alucic) [inline src/Component/Video/Video.php]: Here is the similar thing, I'm not sure what's correct, function signature or assertion, but they are out of sync

**Comment by alucic** [comment]
:+1:

**Comment by localheinz** [comment]
Thank you, @alucic!

**Pull request merged**